### PR TITLE
Commit CanvasItem state only if it changed

### LINKED ISSUE #44378: Snapped node is not moved, but scene is modified after slight drag
**Godot version:**

3.2.4 beta4

**Issue description:**

2uW3oFJwEO

**Steps to reproduce:**
1. Create a 2D node (Sprite for best effect)
2. Enable grid snapping and set it to some relatively high value
3. Click on your node and drag it very gently few pixels
4. When you release button, the node will stay in place (because it's snapped), but Godot will still register it as movement and mark your scene as modified

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #44378

### CODE REVIEW

File: editor/plugins/canvas_item_editor_plugin.cpp [hunk 1]
```diff
@@ -963,8 +963,33 @@ void CanvasItemEditor::_restore_canvas_item_state(List<CanvasItem *> p_canvas_it
...
+
+		// Can't compare Dictionaries with ==.
+		bool different = false;
+		Array keys = new_state.keys();
+		for (int i = 0; i < keys.size(); i++) {
+			if (old_state.get(keys[i], Variant()) != new_state[keys[i]]) {
+				different = true;
+				break;
+			}
+		}
```
Comment (akien-mga) [inline editor/plugins/canvas_item_editor_plugin.cpp:980]: Might be faster and simpler to compare `.hash()`?

Comment (akien-mga) [reply editor/plugins/canvas_item_editor_plugin.cpp:980]:
```gdscript
var dict1 = {
	"a": 0,
}

var dict2 = {
	"a": 0,
}

func _ready():
	print(dict1 == dict2)  # False
	print(dict1.hash() == dict2.hash())  # True
```

Comment (KoBeWi) [reply editor/plugins/canvas_item_editor_plugin.cpp]: Right, I forgot about this...

**Review by groud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4.

**Pull request merged**